PrismaDB-QueryAST/DML/Expressions/AllColumns.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Data;
+
+namespace PrismaDB.QueryAST.DML
+{
+    public class AllColumns : Expression
+    {
+        public TableRef Table;
+
+        public AllColumns(string tableName)
+        {
+            setValue(tableName);
+        }
+
+        public AllColumns()
+            : this("")
+        { }
+
+        public AllColumns(TableRef table)
+            : this()
+        {
+            Table = table.Clone();
+        }
+
+        public override void setValue(params object[] value)
+        {
+            Parent = null;
+            Table = new TableRef((string)value[0]);
+        }
+
+        public override object Clone()
+        {
+            var clone = new AllColumns(Table);
+
+            return clone;
+        }
+
+        public override object Eval(DataRow r)
+        {
+            return r[ToString()];
+        }
+
+        public override List<ColumnRef> GetColumns()
+        {
+            return new List<ColumnRef>();
```

Comment:
And here we have an interesting moment: it should return all columns of a table but that’s something only Core knows about in the schema data.

Reply:
Yea, Core will have to do that when it encounters this. Should we expose a list of columns for Core to access? Or just let Core add the right columns into the encQuery that is going to be executed.

Reply:
Yea, i guess we can throw an exception, and fast track check will fail, and go through the normal process

Reply:
Or, we check the query for AllColumns and skip fast track

Reply:
And determining whether it is a fast track or not also depends on the columns. If it returns the empty list, it could be falsely attributed to fast track. So before the check for a fast track core should substitute star with actual list of columns for the table and then proceed as usual.

Reply:
Done.